corelib/src/libs/SireMove/openmmfrenergyst.cpp

```diff
@@ -2834,47 +2834,52 @@ void OpenMMFrEnergyST::initialise()
 
     if (UseLink_flag == true)
     {
-        Molecule molecule = moleculegroup.moleculeAt(0).molecule();
 
-        bool haslinkinfo = molecule.hasProperty("linkbonds");
-
-        if (haslinkinfo)
+        for (int i = 0; i < nmols; i++)
         {
-            std::vector<double> custom_bond_link_par(3);
-
-            Properties linkprop = molecule.property("linkbonds").asA<Properties>();
 
-            int nlinks = linkprop.property(QString("nbondlinks")).asA<VariantProperty>().toInt();
-
-            if (Debug)
-                qDebug() << "Number of constraint links = " << nlinks;
+            Molecule molecule = moleculegroup.moleculeAt(i).molecule();
+            bool haslinkinfo = molecule.hasProperty("linkbonds");
 
-            for (int i = 0; i < nlinks; i++)
+            if (haslinkinfo)
             {
-                int atomnum0 = linkprop.property(QString("AtomNum0(%1)").arg(i)).asA<VariantProperty>().toInt();
-                int atomnum1 = linkprop.property(QString("AtomNum1(%1)").arg(i)).asA<VariantProperty>().toInt();
-                double reql = linkprop.property(QString("reql(%1)").arg(i)).asA<VariantProperty>().toDouble();
-                double kl = linkprop.property(QString("kl(%1)").arg(i)).asA<VariantProperty>().toDouble();
-                double dl = linkprop.property(QString("dl(%1)").arg(i)).asA<VariantProperty>().toDouble();
+                std::vector<double> custom_bond_link_par(3);
```

Comment:
For consistency with the rest of the code I'd use Qt containers, i.e. use `QVector<double>` rather than `std::vector<double>`.